Cannot build in macOS 11.6 (x86_64)

## Build script

```bash
    git clone --depth=1 https://github.com/intel/isa-l_crypto.git
    pushd isa-l_crypto
    ./autogen.sh
    CFLAGS="-g -O2 -fPIC" ./configure --target=darwin
    make -j"$(sysctl -n hw.ncpu)"
    make install
    popd
```

## Build log

```
Cloning into 'isa-l_crypto'...
remote: Enumerating objects: 540, done.
remote: Counting objects: 100% (540/540), done.
remote: Compressing objects: 100% (418/418), done.
remote: Total 540 (delta 347), reused 183 (delta 119), pack-reused 0
Receiving objects: 100% (540/540), 727.53 KiB | 9.97 MiB/s, done.
Resolving deltas: 100% (347/347), done.

glibtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
glibtoolize: linking file 'build-aux/ltmain.sh'
glibtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.ac,
glibtoolize: and rerunning glibtoolize and aclocal.
glibtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
configure.ac:50: warning: The macro `AC_PROG_CC_STDC' is obsolete.
configure.ac:50: You should run autoupdate.
./lib/autoconf/c.m4:1666: AC_PROG_CC_STDC is expanded from...
configure.ac:50: the top level
configure.ac:68: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:68: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:68: the top level
configure.ac:23: installing 'build-aux/compile'
configure.ac:25: installing 'build-aux/config.guess'
configure.ac:25: installing 'build-aux/config.sub'
configure.ac:12: installing 'build-aux/install-sh'
configure.ac:12: installing 'build-aux/missing'
Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
clang: error: unsupported option '-print-multi-os-directory'
clang: error: no input files

----------------------------------------------------------------
Initialized build system. For a common configuration please run:
----------------------------------------------------------------

./configure --prefix=/usr --libdir=/usr/lib

checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... ./build-aux/install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... rm: conftest.dSYM: is a directory
yes
checking dependency style of gcc... gcc3
checking dependency style of gcc... gcc3
checking build system type... x86_64-apple-darwin20.6.0
checking host system type... x86_64-apple-darwin20.6.0
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking whether make supports nested variables... (cached) yes
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /Applications/Xcode_13.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode_13.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... rm: conftest.dSYM: is a directory
BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 786432
checking how to convert x86_64-apple-darwin20.6.0 file names to x86_64-apple-darwin20.6.0 format... func_convert_file_noop
checking how to convert x86_64-apple-darwin20.6.0 file names to toolchain format... func_convert_file_noop
checking for /Applications/Xcode_13.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... rm: conftest.dSYM: is a directory
no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... rm: conftest.dSYM: is a directory
ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Applications/Xcode_13.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin20.6.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for a sed that does not truncate output... (cached) /usr/bin/sed
checking whether Intel CET is enabled... no
checking for yasm... yes
checking for modern yasm... yes
checking for optional yasm AVX512 support... no
checking for optional yasm SHA-NI support... no
checking for nasm... no
no nasm
checking for optional nasm SHA-NI support... no
Assembler does not understand SHANI opcodes.  Consider upgrading for best performance.
Assembler does not understand AVX512 opcodes.  Consider upgrading for best performance.
Using yasm args target "darwin" "-f macho64 --prefix=_ "
checking for limits.h... yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for inline... inline
checking for size_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for uint8_t... yes
checking for GNU libc compatible malloc... yes
checking for memmove... yes
checking for memset... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libisal_crypto.pc
config.status: executing depfiles commands
config.status: executing libtool commands

        isa-l_crypto 2.24.0
        =====

        prefix:                 /usr
        sysconfdir:             ${prefix}/etc
        libdir:                 ${exec_prefix}/lib
        includedir:             ${prefix}/include

        compiler:               gcc
        cflags:                 -g -O2 -fPIC
        ldflags:                

        debug:                  no

Building isa-l_crypto.h
/Applications/Xcode_13.1.app/Contents/Developer/usr/bin/make --no-print-directory all-am
  MKTMP    sha1_mb/sha1_mb_mgr_submit_sse.s
  MKTMP    sha1_mb/sha1_mb_mgr_submit_avx.s
  MKTMP    sha1_mb/sha1_mb_mgr_submit_avx2.s
  MKTMP    sha1_mb/sha1_mb_mgr_flush_sse.s
  MKTMP    sha1_mb/sha1_mb_mgr_flush_avx.s
  MKTMP    sha1_mb/sha1_mb_mgr_flush_avx2.s
  MKTMP    sha1_mb/sha1_mb_x4_sse.s
  MKTMP    sha1_mb/sha1_mb_x4_avx.s
  MKTMP    sha1_mb/sha1_mb_x8_avx2.s
  MKTMP    sha1_mb/sha1_multibinary.s
  CC       sha1_mb/sha1_ctx_avx512.lo
  CC       sha1_mb/sha1_mb_mgr_init_avx512.lo
  MKTMP    sha1_mb/sha1_mb_mgr_submit_avx512.s
  MKTMP    sha1_mb/sha1_mb_mgr_flush_avx512.s
  MKTMP    sha1_mb/sha1_mb_x16_avx512.s
  MKTMP    sha1_mb/sha1_opt_x1.s
  MKTMP    sha1_mb/sha1_ni_x1.s
  MKTMP    sha1_mb/sha1_ni_x2.s
  CC       sha1_mb/sha1_ctx_sse_ni.lo
  CC       sha1_mb/sha1_ctx_avx512_ni.lo
  MKTMP    sha1_mb/sha1_mb_mgr_submit_sse_ni.s
  MKTMP    sha1_mb/sha1_mb_mgr_flush_sse_ni.s
  MKTMP    sha1_mb/sha1_mb_mgr_flush_avx512_ni.s
  MKTMP    mh_sha1/mh_sha1_multibinary.s
  MKTMP    mh_sha1/mh_sha1_block_sse.s
  MKTMP    mh_sha1/mh_sha1_block_avx.s
  MKTMP    mh_sha1/mh_sha1_block_avx2.s
  MKTMP    mh_sha1/mh_sha1_block_avx512.s
  CC       mh_sha1/mh_sha1_avx512.lo
  MKTMP    md5_mb/md5_mb_mgr_submit_sse.s
  MKTMP    md5_mb/md5_mb_mgr_submit_avx.s
  MKTMP    md5_mb/md5_mb_mgr_submit_avx2.s
  MKTMP    md5_mb/md5_mb_mgr_flush_sse.s
  MKTMP    md5_mb/md5_mb_mgr_flush_avx.s
  MKTMP    md5_mb/md5_mb_mgr_flush_avx2.s
  MKTMP    md5_mb/md5_mb_x4x2_sse.s
  MKTMP    md5_mb/md5_mb_x4x2_avx.s
  MKTMP    md5_mb/md5_mb_x8x2_avx2.s
  MKTMP    md5_mb/md5_multibinary.s
  MKTMP    md5_mb/md5_mb_mgr_submit_avx512.s
  MKTMP    md5_mb/md5_mb_mgr_flush_avx512.s
  MKTMP    md5_mb/md5_mb_x16x2_avx512.s
  CC       md5_mb/md5_ctx_avx512.lo
  MKTMP    sha256_mb/sha256_mb_mgr_submit_sse.s
  MKTMP    sha256_mb/sha256_mb_mgr_submit_avx.s
  MKTMP    sha256_mb/sha256_mb_mgr_submit_avx2.s
  MKTMP    sha256_mb/sha256_mb_mgr_flush_sse.s
  MKTMP    sha256_mb/sha256_mb_mgr_flush_avx.s
  MKTMP    sha256_mb/sha256_mb_mgr_flush_avx2.s
  MKTMP    sha256_mb/sha256_mb_x4_sse.s
  MKTMP    sha256_mb/sha256_mb_x4_avx.s
  MKTMP    sha256_mb/sha256_mb_x8_avx2.s
  MKTMP    sha256_mb/sha256_multibinary.s
  CC       sha256_mb/sha256_ctx_avx512.lo
  CC       sha256_mb/sha256_mb_mgr_init_avx512.lo
md5_mb/md5_ctx_avx512.c:31:16: error: unterminated '#pragma clang attribute push' at end of file
# pragma clang attribute push (__attribute__((target("avx2"))), apply_to=function)
               ^
1 error generated.
make[1]: *** [md5_mb/md5_ctx_avx512.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
rm sha1_mb/sha1_mb_x16_avx512.s md5_mb/md5_mb_x4x2_sse.s sha1_mb/sha1_mb_x4_avx.s sha256_mb/sha256_mb_mgr_flush_avx2.s sha1_mb/sha1_mb_mgr_flush_avx2.s mh_sha1/mh_sha1_block_avx512.s sha256_mb/sha256_mb_mgr_flush_sse.s sha256_mb/sha256_mb_x4_sse.s md5_mb/md5_mb_x8x2_avx2.s sha1_mb/sha1_mb_mgr_submit_sse.s md5_mb/md5_mb_mgr_submit_avx.s md5_mb/md5_mb_mgr_submit_avx2.s sha1_mb/sha1_multibinary.s sha1_mb/sha1_mb_mgr_submit_avx.s sha1_mb/sha1_ni_x2.s sha1_mb/sha1_mb_x4_sse.s sha256_mb/sha256_mb_mgr_submit_avx.s mh_sha1/mh_sha1_block_avx.s sha256_mb/sha256_mb_mgr_submit_sse.s sha1_mb/sha1_opt_x1.s sha256_mb/sha256_mb_x4_avx.s sha1_mb/sha1_mb_mgr_flush_avx512.s md5_mb/md5_mb_mgr_flush_avx2.s sha1_mb/sha1_mb_mgr_flush_avx.s md5_mb/md5_mb_mgr_flush_sse.s sha1_mb/sha1_mb_mgr_submit_avx512.s md5_mb/md5_mb_mgr_submit_sse.s sha1_mb/sha1_mb_mgr_submit_avx2.s sha1_mb/sha1_mb_mgr_flush_avx512_ni.s sha1_mb/sha1_mb_mgr_flush_sse_ni.s sha1_mb/sha1_mb_x8_avx2.s sha256_mb/sha256_mb_mgr_submit_avx2.s mh_sha1/mh_sha1_block_avx2.s sha1_mb/sha1_mb_mgr_flush_sse.s sha1_mb/sha1_mb_mgr_submit_sse_ni.s sha256_mb/sha256_mb_x8_avx2.s md5_mb/md5_mb_x4x2_avx.s md5_mb/md5_multibinary.s md5_mb/md5_mb_x16x2_avx512.s mh_sha1/mh_sha1_multibinary.s sha256_mb/sha256_mb_mgr_flush_avx.s sha256_mb/sha256_multibinary.s sha1_mb/sha1_ni_x1.s md5_mb/md5_mb_mgr_flush_avx512.s mh_sha1/mh_sha1_block_sse.s md5_mb/md5_mb_mgr_flush_avx.s md5_mb/md5_mb_mgr_submit_avx512.s
make: *** [all] Error 2

```

```
md5_mb/md5_ctx_avx512.c:31:16: error: unterminated '#pragma clang attribute push' at end of file
# pragma clang attribute push (__attribute__((target("avx2"))), apply_to=function)
               ^
1 error generated.
make[1]: *** [md5_mb/md5_ctx_avx512.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
rm sha1_mb/sha1_mb_x16_avx512.s md5_mb/md5_mb_x4x2_sse.s sha1_mb/sha1_mb_x4_avx.s sha256_mb/sha256_mb_mgr_flush_avx2.s sha1_mb/sha1_mb_mgr_flush_avx2.s mh_sha1/mh_sha1_block_avx512.s sha256_mb/sha256_mb_mgr_flush_sse.s sha256_mb/sha256_mb_x4_sse.s md5_mb/md5_mb_x8x2_avx2.s sha1_mb/sha1_mb_mgr_submit_sse.s md5_mb/md5_mb_mgr_submit_avx.s md5_mb/md5_mb_mgr_submit_avx2.s sha1_mb/sha1_multibinary.s sha1_mb/sha1_mb_mgr_submit_avx.s sha1_mb/sha1_ni_x2.s sha1_mb/sha1_mb_x4_sse.s sha256_mb/sha256_mb_mgr_submit_avx.s mh_sha1/mh_sha1_block_avx.s sha256_mb/sha256_mb_mgr_submit_sse.s sha1_mb/sha1_opt_x1.s sha256_mb/sha256_mb_x4_avx.s sha1_mb/sha1_mb_mgr_flush_avx512.s md5_mb/md5_mb_mgr_flush_avx2.s sha1_mb/sha1_mb_mgr_flush_avx.s md5_mb/md5_mb_mgr_flush_sse.s sha1_mb/sha1_mb_mgr_submit_avx512.s md5_mb/md5_mb_mgr_submit_sse.s sha1_mb/sha1_mb_mgr_submit_avx2.s sha1_mb/sha1_mb_mgr_flush_avx512_ni.s sha1_mb/sha1_mb_mgr_flush_sse_ni.s sha1_mb/sha1_mb_x8_avx2.s sha256_mb/sha256_mb_mgr_submit_avx2.s mh_sha1/mh_sha1_block_avx2.s sha1_mb/sha1_mb_mgr_flush_sse.s sha1_mb/sha1_mb_mgr_submit_sse_ni.s sha256_mb/sha256_mb_x8_avx2.s md5_mb/md5_mb_x4x2_avx.s md5_mb/md5_multibinary.s md5_mb/md5_mb_x16x2_avx512.s mh_sha1/mh_sha1_multibinary.s sha256_mb/sha256_mb_mgr_flush_avx.s sha256_mb/sha256_multibinary.s sha1_mb/sha1_ni_x1.s md5_mb/md5_mb_mgr_flush_avx512.s mh_sha1/mh_sha1_block_sse.s md5_mb/md5_mb_mgr_flush_avx.s md5_mb/md5_mb_mgr_submit_avx512.s
make: *** [all] Error 2

$ sw_vers
ProductName:	macOS
ProductVersion:	11.6.1
BuildVersion:	20G224

$ clang --version
Apple clang version 13.0.0 (clang-1300.0.29.3)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Applications/Xcode_13.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

--- a/md5_mb/md5_ctx_avx512.c
+++ b/md5_mb/md5_ctx_avx512.c
@@ -260,8 +260,8 @@ struct slver md5_ctx_mgr_submit_avx512_slver = { 0x018d, 0x00, 0x06 };
 struct slver md5_ctx_mgr_flush_avx512_slver_0600018e;
 struct slver md5_ctx_mgr_flush_avx512_slver = { 0x018e, 0x00, 0x06 };
 
+#endif // HAVE_AS_KNOWS_AVX512
+
 #if defined(__clang__)
 # pragma clang attribute pop
 #endif
-
-#endif // HAVE_AS_KNOWS_AVX512

